<launch>

<param name="use_sim_time" value="true" />

<param name="Xtion_camera" value="true" />

<!-- Select classifier "RGB" or "DepthJet" -->
<param name="classifier_type" value="DepthJet" />

<!-- load parameters for mobility aids detector -->
<rosparam file="$(find hospital_people_detector)/config/hospital_params.yaml" command="load" />

<arg name="arg_camera_height" default="1.1" />
<param name="camera_height" value="$(arg arg_camera_height)" />

<param name="save_images" value="false"/>
<param name="directory_for_images" value="$(find hospital_people_detector)/saved_images/" />

<!-- Camera topics -->
<param name="rgb_image_topic" value="/camera/rgb/image_raw" />
<param name="depth_image_topic" value="/camera/depth_registered/image_raw" />
<param name="camera_info_topic" value="/camera/depth_registered/camera_info" />

<!-- Frames -->
<param name="fixed_frame" value="/odom" />
<param name="camera_frame" value="/camera_depth_optical_frame" />
<param name="base_frame" value="/base_link" />

<!-- Fast R-CNN directories -->
<param name="caffe_directory" value="/home/andres/fast-rcnn/caffe-fast-rcnn" />
<param name="fast_rcnn_directory" value="/home/andres/fast-rcnn" />

<!-- Classifier files -->
<param name="hospital_prototxt" value="$(find hospital_people_detector)/trained_models/deploy_rgb.prototxt" />
<param name="DepthJet_caffemodel" value="$(find hospital_people_detector)/trained_models/DepthJet.caffemodel" />
<param name="RGB_caffemodel" value="$(find hospital_people_detector)/trained_models/RGB.caffemodel" />

<!-- Simulate tf -->
<node pkg="tf" type="static_transform_publisher" name="tf_link_00" args="0 0 0 0 0 0 /odom /base_link 33" />
<node pkg="tf" type="static_transform_publisher" name="tf_link_01" args="0.2 0 1.1 0 0.1 0 /base_link /camera_link 33" />

<!-- Run nodes -->
<node pkg="hospital_people_detector" type="proposals_generation" name="proposals_generation" output="screen"/>
<node pkg="hospital_people_detector" type="detection_tracking.py" name="detection_tracking" output="screen"/>

</launch>
